docs/sources/reference/stdlib/targets.md

```diff
@@ -0,0 +1,38 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/stdlib/targets/
+description: Learn about targets functions
+menuTitle: targets
+title: targets
+---
+
+# targets
+
+The `targets` namespace contains functions related to `list(map(string))` arguments.
+They are often used by [prometheus][prom-comp] and [discovery][disc-comp] components.
+Refer to [Compatible components][] for a full list of components which export and consume targets.
+
+[prom-comp]: ../components/prometheus/
+[disc-comp]: ../components/discovery/
+[Compatible components]: ../compatibility/#targets
+
+## targets.merge
+
+The `targets.inner_join` function allows you to join two arrays containing maps if certain keys have matching values in both maps.
+It takes three inputs:
+
+* The first two inputs are a of type `list(map(string))`. The keys of the map are strings. 
+  The value for each key could have any Alloy type such as a string, integer, map, or a capsule.
+* The third input is an array containing strings. The strings are the keys whose value has to match for maps to be joined.
```

Comment:
Currently, the "third input" assumes both targets have a label with the same value. I wonder if it's worth changing the third input to be an array of arrays, where we can set different label names in the LHS and RHS targets. E.g.:

```
[["lhs_lbl_name"], ["rhs_lbl_name"], ["lhs_lbl_name_2"], ["rhs_lbl_name_2"]]
```

This could reduce the amount of relabels a user would have to do. But I don't know if it's really worth the extra complexity. The users probably have to relabel anyway.

Reply:
I would say it's not worth the complexity. If there's a demand for this, we can create new function rather than impose extra complexity on everyone who may not even need it.